server/document.js

```diff
@@ -170,47 +159,36 @@ export class NextScript extends Component {
       return <script
         key={file}
         src={`${assetPrefix}/_next/${file}`}
-        nonce={this.props.nonce}
         async
       />
     })
   }
 
-  static getInlineScriptSource (documentProps) {
-    const { __NEXT_DATA__ } = documentProps
-    const { page, pathname } = __NEXT_DATA__
-
-    return `
-      __NEXT_DATA__ = ${htmlescape(__NEXT_DATA__)}
-      module={}
-      __NEXT_LOADED_PAGES__ = []
-
-      __NEXT_REGISTER_PAGE = function (route, fn) {
-        __NEXT_LOADED_PAGES__.push({ route: route, fn: fn })
-      }${page === '_error' ? `
-
-      __NEXT_REGISTER_PAGE(${htmlescape(pathname)}, function() {
-        var error = new Error('Page does not exist: ${htmlescape(pathname)}')
-        error.statusCode = 404
-
-        return { error: error }
-      })`: ''}
-    `
-  }
-
   render () {
     const { staticMarkup, assetPrefix, devFiles, __NEXT_DATA__ } = this.context._documentProps
     const { page, pathname, buildId } = __NEXT_DATA__
     const pagePathname = getPagePathname(pathname)
 
+    __NEXT_DATA__.cleanPathname = htmlescape(__NEXT_DATA__.pathname);
```

Comment:
What is the possible attack vector here, assuming that's the reason you're escaping the pathname 🤔 `pathname` is already part of `__NEXT_DATA__`, and `__NEXT_DATA__` gets htmlescaped too

Reply:
I didn’t want to have to bundle htmlescape into the the bootstrap file. I just escape the value early: https://github.com/zeit/next.js/blob/canary/server/document.js#L192